### PR TITLE
JOINDIN-564 Logout should leave me on the same page

### PR DESCRIPTION
**Pull request by kazu9su**
Hi team,
This is my first contribution of this project.
I tried to solve below issue.
https://joindin.jira.com/browse/JOINDIN-564

Thanks!
Kazu

### CODE REVIEW

**Comment by akrabat** [comment]
ok to test

**Comment by lornajane** [comment]
The build for this pull request has failed, you can view the output of the build (http://jenkins.joind.in/job/web2-pr-build/255/console) to find the exact error messages, but they are all coding standards fixes.  We use PSR2 so please correct the issues (mostly whitespace) and when the build passes, we'll take another look at the patch.

**Comment by kazu9su** [comment]
ok I'll check it and fix coding style.
The other hand,  I found an error in self-test.
So, I'll fix it too and send new pull request.
Thanks!

**Comment by akrabat** [comment]
There's no need to close this PR. You can make the changes on your `JOINDIN-564` branch and push them. Github will notice and update this PR appropriately.

**Comment by kazu9su** [comment]
Thanks for your advice, Rob:)
So please wait a moment to push code...

**Comment by kazu9su** [comment]
I pushed my changes to JOINDIN-564 branch.
Please confirm that.

Thanks!

**Comment by akrabat** [comment]
ok to test

**Comment by akrabat** [comment]
One error left: http://jenkins.joind.in/job/web2-pr-build/258/console

**Comment by kazu9su** [comment]
I pulled changes again

**Comment by akrabat** [comment]
I think that passing a `redirect` variable to the view in every single controller action isn't the right approach. 

It would be better to either add a variable called `currentPath` to the [Twig global context](http://twig.sensiolabs.org/doc/advanced.html#globals) or by using a [Twig extension function](http://twig.sensiolabs.org/doc/advanced.html#functions) accessed as `{{ currentPath() }}`.

Also, in `layout.html.twig`, I don't want to see  `?redirect=` when there isn't a path set be set.

**Comment by kazu9su** [comment]
Thank you for your review.
OK, I'll refactoring my code

**Comment by kazu9su** [comment]
I pushed all my changes.
Please confirm it again.
Thanks!

File: app/src/Search/SearchController.php [hunk 1]
```diff
@@ -83,10 +83,10 @@ public function searchEvents()
...
             'Event/search.html.twig',
             array(
-                'events'  => $events,
-                'page'    => $page,
-                'keyword' => $keyword,
-                'tag'     => $tag
+                'events'   => $events,
+                'page'     => $page,
+                'keyword'  => $keyword,
+                'tag'      => $tag,
```
Comment (heiglandreas) [inline app/src/Search/SearchController.php]: Why the extra space?

File: app/src/User/UserController.php [hunk 2]
```diff
@@ -122,8 +124,8 @@ public function register()
         $this->render(
             'User/register.html.twig',
             array(
-                'form' => $form->createView(),
-            )
+                'form'     => $form->createView(),
```
Comment (heiglandreas) [inline app/src/User/UserController.php]: Why the extra space?

File: app/src/User/UserController.php [hunk 3]
```diff
@@ -604,7 +609,7 @@ public function remindUsername()
         $this->render(
             'User/username-reminder.html.twig',
             array(
-                'form' => $form->createView(),
+                'form'     => $form->createView(),
```
Comment (heiglandreas) [inline app/src/User/UserController.php]: And again: why the extra space? :wink:

**Comment by akrabat** [comment]
I think we're definitely on the right track now. 

Next things:

1. Please revert all unnecessary changes in the controllers, including the change to the Login link as this issue is for log out.
2. It doesn't actually work. When I click log out, I always end up on the home page.

**Comment by kazu9su** [comment]
OK so at first, I revert unnecessary changes.

**Comment by kazu9su** [comment]
and about 2nd problem.
I cannot confirm( or sorry but cannot understand..) this problem.
Could you explain in detail?

Thanks a lot!

**Comment by heiglandreas** [comment]
I can reproduce what @akrabat describes. I open up the joind.in-website (web2) and go to any events detail-page. I then click onto the login-button and login. I'd expect that, after a successfull login, I'd be redirected to the events detail-page I've been on before. Instead I see the "Hot Events" and the URL reads ```<server-address>:<port>```

The same happens when I'm on any detail-page and click "logout". I'd expect to see the detail-page again, but instead I see the "Hot Events". The URL reads ```<server-address>:<port>/?redirect=/event/ruby-meetup```.

**Comment by akrabat** [comment]
Note that this issue & PR should only be for handling the logout link.

**Comment by kazu9su** [comment]
Thank you all.

Okay, I'll fix it!

**Comment by kazu9su** [comment]
Hi all,
I've fixed the problem.
Please confirm it.

By the way, there is something I'm worrying about a little.
The route "/user/<username>" should have a login filter.
How about it?

Thanks all!

**Comment by akrabat** [comment]
Thanks kazu9su, I'll review the changes when I can.

/user/username certainly shouldn't have a login guard as it shows useful public information about a given user. e.g. https://m.joind.in/user/akrabat

**Comment by kazu9su** [comment]
Hi akrabat,
I see.
Thanks for your cooperation and kindness!

**Comment by akrabat** [comment]
Committed to master and now live!

Thank you @kazu9su :)

**Pull request closed without merging**